lib/Ptero/Builder/Workflow.pm

```diff
@@ -43,33 +43,29 @@ override 'BUILDARGS' => sub {
     return $params;
 };
 
-sub submit_url {
-    if (exists $ENV{PTERO_WORKFLOW_SUBMIT_URL}) {
-        return $ENV{PTERO_WORKFLOW_SUBMIT_URL};
-    } else {
-        die "environment variable PTERO_WORKFLOW_SUBMIT_URL must be set";
-    }
-}
-
 sub submit {
     my $self = shift;
     my %p = Params::Validate::validate(@_, {
         inputs => {type => HASHREF, optional => 1},
         name => {type => SCALAR, optional => 1},
+        submit_url => {type => SCALAR, optional => 1},
     });
 
+    my $submit_url = $p{submit_url} || $ENV{PTERO_WORKFLOW_SUBMIT_URL} || die
+        "Must specify 'submit_url' or set PTERO_WORKFLOW_EXECUTION_URL env variable.";
```

Comment:
Yea stupid tab completion... the die message is wrong so at least that's less wrong than the other thing.  I'll fix, nice catch Tom!

____
This email message is a private communication. The information transmitted, including attachments, is intended only for the person or entity to which it is addressed and may contain confidential, privileged, and/or proprietary material. Any review, duplication, retransmission, distribution, or other use of, or taking of any action in reliance upon, this information by persons or entities other than the intended recipient is unauthorized by the sender and is prohibited. If you have received this message in error, please contact the sender immediately by return email and delete the original message from all computer systems. Thank you.